transformer_engine/pytorch/transformer.py

```diff
@@ -82,9 +84,6 @@ def __init__(
         norm_factor: float,
         attention_dropout: float = 0.0,
         attention_dropout_ctx: Optional[Callable] = nullcontext,
-        layer_number: Optional[int] = None,
```

Comment:
I'm not sure about removal of this layer number. Even if that is just a numerical hack, we should probably do it (just automatically) when needed, and then this would be needed.

Reply:
It is a completely unused trick that was needed at a much earlier time to train large models in fp16. No other attention implementation (including FA) has this. MLM team said that it is no longer needed too.